lib/xapi-fdcaps/test/properties.t

```diff
@@ -0,0 +1,23 @@
+Check that we get compile errors when trying to use a read-only or write-only property with the opposite operation:
+
+  $ cat >t.ml <<'EOF'
+  > open Xapi_fdcaps.Properties
+  > let _ = as_readable (make `wronly `reg) in ()
+  > EOF
+  $ ocamlfind ocamlc -package xapi-fdcaps -c t.ml
+  File "t.ml", line 2, characters 40-42:
+  2 | let _ = as_readable (make `wronly `reg) in ()
```

Comment:
Would it be an option to grep for Syntax error so it can be compiler independent?

Reply:
Actually that is the wrong error message, thanks for spotting.
The real error is something longer like:
```
Error: This expression has type
           ([< Xapi_fdcaps.Properties.rw > `rdonly ] as 'a,
            [< Xapi_fdcaps.Properties.kind > `reg ] as 'b)
           Xapi_fdcaps.Properties.t = ('a, 'b) Xapi_fdcaps.Properties.props
         but an expression was expected of type
           ([< Xapi_fdcaps.Properties.writable ] as 'c,
            [< Xapi_fdcaps.Properties.kind ] as 'd)
           Xapi_fdcaps.Properties.t = ('c, 'd) Xapi_fdcaps.Properties.props
         Type
           [< Xapi_fdcaps.Properties.rw > `rdonly ] =
             [< `rdonly | `rdwr | `wronly > `rdonly ]
         is not compatible with type
           [< Xapi_fdcaps.Properties.writable ] = [< `rdwr | `wronly ]
         The second variant type does not allow tag(s) `rdonly
```

I pushed a fixup

Reply:
Maybe I could pipe the output to tail and match only on the last line? That is unlikely to change between versions

Reply:
that might work, yeah

Reply:
I used `tail -n 1` now, and removed the OCaml version constraint from the tests.

Reply:
I also tried to use `set -o pipefail` but that failed on the CI, probably due to a non-bash shell. Anyway we've got the last line of the error output, so if it suddenly turns out that there is no error then that will change and cause a test failure, so it should be fine without pipefail in this case